cms/djangoapps/contentstore/views/preview.py

```diff
@@ -133,6 +139,36 @@ def layout_asides(self, block, context, frag, view_name, aside_frag_fns):
         result.add_content(frag.content)
         return result
 
+    def service(self, block, service_name):
+        """
+        Runtime-specific override for the XBlock service manager.  If a service is not currently
+        instantiated and is declared as a critical requirement, an attempt is made to load the
+        module.
+
+        Arguments:
+            block (an XBlock): this block's class will be examined for service
+                decorators.
+            service_name (string): the name of the service requested.
+
+        Returns:
+            An object implementing the requested service, or None.
+        """
+        declaration = block.service_declaration(service_name)
+        if declaration is None:
+            raise NoSuchServiceError("Service {!r} was not requested.".format(service_name))
+
+        service = self._services.get(service_name)
+        if service is None and declaration == "need":
+            service_map = {
+                'i18n': ModuleI18nService,
+            }
+            try:
+                service = service_map[service_name](block)
+            except KeyError:
+                raise NoSuchServiceError("Service {!r} is not available.".format(service_name))
+
+        return service
```

Comment:
Rather than re-implementing the whole method here, you might just add the dynamic initialization code, and then call into super to get the statically initialized services.